nvdaHelper/localWin10/uwpOcr.cpp

```diff
@@ -13,78 +13,82 @@ This license can be found at:
 http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
 */
 
-#include <collection.h>
-#include <ppltasks.h>
-#include <wrl.h>
-#include <robuffer.h>
+#include <winrt/Windows.Storage.Streams.h>
+#include <winrt/Windows.Media.Ocr.h>
+#include <winrt/Windows.Foundation.Collections.h>
+#include <winrt/Windows.Globalization.h>
+#include <winrt/Windows.Graphics.Imaging.h>
+#include <winrt/Windows.Data.Json.h>
 #include <windows.h>
 #include <cstring>
 #include <common/log.h>
-#include "utils.h"
 #include "uwpOcr.h"
 
 using namespace std;
-using namespace Platform;
-using namespace concurrency;
-using namespace Windows::Storage::Streams;
-using namespace Microsoft::WRL;
-using namespace Windows::Media::Ocr;
-using namespace Windows::Foundation::Collections;
-using namespace Windows::Globalization;
-using namespace Windows::Graphics::Imaging;
-using namespace Windows::Data::Json;
+using namespace winrt;
+using namespace winrt::Windows::Storage::Streams;
+using namespace winrt::Windows::Media::Ocr;
+using namespace winrt::Windows::Foundation::Collections;
+using namespace winrt::Windows::Globalization;
+using namespace winrt::Windows::Graphics::Imaging;
+using namespace winrt::Windows::Data::Json;
 
-UwpOcr* __stdcall uwpOcr_initialize(const char16* language, uwpOcr_Callback callback) {
-	auto engine = OcrEngine::TryCreateFromLanguage(ref new Language(ref new String(language)));
-	if (!engine)
+UwpOcr::UwpOcr(OcrEngine const& engine, uwpOcr_Callback callback) : engine(engine), callback(callback) { }
+
+UwpOcr* __stdcall uwpOcr_initialize(const wchar_t* language, uwpOcr_Callback callback) {
+	auto engine = OcrEngine::TryCreateFromLanguage(Language{ language });
+	if (!engine) {
 		return nullptr;
-	auto instance = new UwpOcr;
-	instance->engine = engine;
-	instance->callback = callback;
+	}
+	auto instance = new UwpOcr{ engine, callback };

```

Comment:
This isn't very safe (potential memory leaks, loss of ownership), this should use a smart pointer instead.
We have the opportunity to change this design for the 2022.1 release, or we can maintain the design and (I think) return a ccomPtr to Python.

Reply:
We can follow up on this via new issue: https://github.com/nvaccess/nvda/issues/13253